### DISCUSSION TITLE
Server Error 500 (data-import combined with Rabobank JSON file)

### DISCUSSION THREAD

**Discussion by dschigt** [firefly-iii/firefly-iii#6517] [General]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely me, not you](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Hi,

After starting the data-importer, I am able to read/process a CSV file.
However, upon providing a JSON file (this case Rabobank) it immediately throws an 500 | SERVER ERROR.

The few log files I was able to read did not yield a clue...
Honestly, I have no idea what is happening and I have been scratching my head for a few hours now to get it working...

Any help in this would be very much appreciated.

Best regards,

Detlev Schigt

### Expected behaviour

What I would expect is that after pressing "Next ->" button, the CSV file is being processed with guidance of the JSON file

### Debug information

Actually, I am unable to provide this info as I do not know where /debug is located...

### Steps to reproduce

1. Start data-importer
2. Provide CSV file (as exported from Rabobank site)
3. Provide Rabobank JSON file (as provided at Firefly-III GirHub)

### Additional info

The Firefly-III installation is self hosted (Ubuntu 22.04).
Firefly-III version is: 5.7.12

**Comment by JC5** [comment]
Without the error there isn't much I can do. Try making the CSV file smaller by removing lines. I'm curious to see if it happens with a specific line.

**Reply by dschigt** [reply]
Hi JC5,

Thanks for the swift response. In the meantime I have been able to dig some log files/entries. See below:

**Apache2 FIDI/data-importer (/var/log/apache2/fidi.lan-access_log)**
192.168.2.12 - - [07/Oct/2022:07:02:54 +0200] "GET / HTTP/1.1" 302 1570
192.168.2.12 - - [07/Oct/2022:07:02:55 +0200] "GET /token HTTP/1.1" 302 4646
192.168.2.12 - - [07/Oct/2022:07:02:55 +0200] "GET / HTTP/1.1" 200 3253
192.168.2.12 - - [07/Oct/2022:07:02:55 +0200] "GET /css/bootstrap-dark.min.css?version=0.9.16 HTTP/1.1" 200 31449
192.168.2.12 - - [07/Oct/2022:07:02:55 +0200] "GET /fa/css/all.min.css HTTP/1.1" 200 13013
192.168.2.12 - - [07/Oct/2022:07:02:55 +0200] "GET /fa/webfonts/fa-solid-900.woff2 HTTP/1.1" 200 79734
192.168.2.12 - - [07/Oct/2022:07:02:55 +0200] "GET /validate/spectre HTTP/1.1" 200 1164
192.168.2.12 - - [07/Oct/2022:07:02:55 +0200] "GET /validate/nordigen HTTP/1.1" 200 1164
192.168.2.12 - - [07/Oct/2022:07:02:56 +0200] "GET /favicon.ico HTTP/1.1" 200 295
192.168.2.12 - - [07/Oct/2022:07:02:55 +0200] "GET /token/validate HTTP/1.1" 200 1175
192.168.2.12 - - [07/Oct/2022:07:02:58 +0200] "POST / HTTP/1.1" 302 1915
192.168.2.12 - - [07/Oct/2022:07:02:58 +0200] "GET /authenticate HTTP/1.1" 302 1574
192.168.2.12 - - [07/Oct/2022:07:02:58 +0200] "GET /upload HTTP/1.1" 200 2604
192.168.2.12 - - [07/Oct/2022:07:03:09 +0200] "POST /upload HTTP/1.1" 302 1575
192.168.2.12 - - [07/Oct/2022:07:03:09 +0200] "GET /upload HTTP/1.1" 302 1624
192.168.2.12 - - [07/Oct/2022:07:03:09 +0200] "GET /import/configure HTTP/1.1" 500 7746

**Apache2 (Firefly-III) (/var/log/apache2/firefly.lan-access_log**
192.168.2.190 - - [07/Oct/2022:07:02:55 +0200] "GET /api/v1/about HTTP/1.1" 200 1047

**Data-Importer application log (/var/www/html/data-importer/storage/logs)**
See attached log file: data-import-2022-10-07.log

**Firefly application log (/var/www/html/firefly-iii/storage/logs**
<neither log file created nor entries exists for this specific upload attempt>

Furthermore, be informed:
1. The CSV I used (WITHOUT an accompanied JSON file contains just a 25 entries. These got imported.
2. Limiting the CSV to 5 transactions (and then re-upload WITH JSON file) returns the same issue. Meaning an error 500.

Finally, find attached a screenshot (20221007 at 071924.png) where I got stuck, after hitting the 'Next ->' button.

Thanks again for any support/assistance in this matter!

**Reply by JC5** [reply]
The attachments get lost when you reply via email. Could you attach them on GitHub itself? Be aware they will be public.

**Comment by dschigt** [comment]
Hi JC5,

Ok - Here a new attempt to attach the mentioned log files. Actually, I uploaded them earlier via this web form...
Anyhow, may be this time they stick better... ;-)

[data-import-2022-10-07.log](https://github.com/firefly-iii/firefly-iii/files/9736969/data-import-2022-10-07.log)
![20221007 at 071924](https://user-images.githubusercontent.com/94546059/194639478-571580cd-5083-4959-b268-a1988a23c853.png)

Best regards,

Detlev Schigt

**Reply by JC5** [reply]
Thanks! That looks like a bug, I'll pick it up!

**Reply by dschigt** [reply]
Hi JC5,

Can you perhaps inform me if the bug has been resolved in the meantime? If so, has it been included in a release, else, when do you perhaps expect it will become available in a release?

Best regards,

Detlev Schigt

**Reply by JC5** [reply]
Yes, this should be resolved in the last version.

**Comment by dschigt** [comment]
Hi Jason,

In the meantime I tested the issue, however, for me it remains. As such I am still not able use the template import function.
Any update/clarification on this perhaps? Or, alternatively, can I help with data/screenshots you would like to see specifically?

Thanks!

Best regards,

Detlev

**Reply by JC5** [reply]
I would like to see the logs of the error :)
